staging/operator-lifecycle-manager/pkg/controller/operators/olm/downstream_csv_namespace_labeler_plugin.go

```diff
@@ -0,0 +1,212 @@
+	"openshift-cloud-controller-manager",
+	"openshift-cloud-controller-manager-operator",
+	"openshift-cloud-credential-operator",
+	"openshift-cloud-network-config-controller",
+	"openshift-cluster-csi-drivers",
+	"openshift-cluster-machine-approver",
+	"openshift-cluster-node-tuning-operator",
+	"openshift-cluster-samples-operator",
+	"openshift-cluster-storage-operator",
+	"openshift-cluster-version",
+	"openshift-config",
+	"openshift-config-managed",
+	"openshift-config-operator",
+	"openshift-console",
+	"openshift-console-operator",
+	"openshift-console-user-settings",
+	"openshift-controller-manager",
+	"openshift-controller-manager-operator",
+	"openshift-dns",
+	"openshift-dns-operator",
+	"openshift-etcd",
+	"openshift-etcd-operator",
+	"openshift-host-network",
+	"openshift-image-registry",
+	"openshift-infra",
+	"openshift-ingress",
+	"openshift-ingress-canary",
+	"openshift-ingress-operator",
+	"openshift-insights",
+	"openshift-kni-infra",
+	"openshift-kube-apiserver",
+	"openshift-kube-apiserver-operator",
+	"openshift-kube-controller-manager",
+	"openshift-kube-controller-manager-operator",
+	"openshift-kube-scheduler",
+	"openshift-kube-scheduler-operator",
+	"openshift-kube-storage-version-migrator",
+	"openshift-kube-storage-version-migrator-operator",
+	"openshift-machine-api",
+	"openshift-machine-config-operator",
+	"openshift-marketplace",
+	"openshift-monitoring",
+	"openshift-multus",
+	"openshift-network-diagnostics",
+	"openshift-network-operator",
+	"openshift-node",
+	"openshift-nutanix-infra",
+	"openshift-oauth-apiserver",
+	"openshift-openstack-infra",
+	"openshift-operator-lifecycle-manager",
+	"openshift-ovirt-infra",
+	"openshift-sdn",
+	"openshift-service-ca",
+	"openshift-service-ca-operator",
+	"openshift-user-workload-monitoring",
+	"openshift-vsphere-infra",
+)
+
+type CSVNamespaceLabelerPlugin struct {
+	operator        *Operator
+	namespaceLister v12.NamespaceLister
+}
+
+func (p *CSVNamespaceLabelerPlugin) Init(ctx context.Context, config *operatorConfig, op *Operator) error {
+	if op == nil {
+		return fmt.Errorf("cannot initialize plugin: operator undefined")
+	}
+
+	p.operator = op
+
+	for _, namespace := range config.watchedNamespaces {
+
+		// create a namespace informer for namespaces that do not include
+		// the label syncer label
+		namespaceInformer := informers.NewSharedInformerFactoryWithOptions(
+			op.opClient.KubernetesInterface(),
+			config.resyncPeriod(),
+			informers.WithNamespace(namespace),
+		).Core().V1().Namespaces()
+
+		if err := op.RegisterInformer(namespaceInformer.Informer()); err != nil {
+			return err
+		}
+		p.namespaceLister = namespaceInformer.Lister()
+
+		// create a new csv informer and prune status to reduce memory footprint
+		csvNamespaceLabelerInformer := cache.NewSharedIndexInformer(
+			pruning.NewListerWatcher(
+				op.client,
+				namespace,
+				func(opts *metav1.ListOptions) {
+					// opts.LabelSelector = fmt.Sprintf("!%s", v1alpha1.CopiedLabelKey)
```

Comment:
Why is this commented out?

Reply:
It was a premature optimization. While it can help reduce the memory footprint, it means a 'not found' error when you try to get it (you might still get csv events from csvs in namespaces that are culled). This felt a bit ambiguous and a potential source of confusion for the next developer. wdyt?

Reply:
sorry, I confused myself here. The reason I commented this one out was because I was getting some strange behavior. I wasn't getting events for a non-openshift-* namespace...? I will try again to make sure though.